document-portal/document-portal.h

```diff
@@ -41,8 +41,8 @@ gboolean validate_fd (int           fd,
 char ** document_add_full (int                      *fd,
                            int                      *parent_dev,
                            int                      *parent_ino,
+                           DocumentAddFullFlags     *documents_flags,
                            int                       n_args,
-                           DocumentAddFullFlags      flags,
                            XdpAppInfo               *app_info,
```

Comment:
This change is so that each fd can get its own set of flags?

An alternative to this flag array would be to let the caller split up their fd list into files and directories, and do separate calls (with one set of flags each) to add them.

Not sure that is better. It would let us avoid this array of flags, but it is certainly more work for the callers.

Reply:
I felt having `document_add_full()` handle it was really simpler on every aspects. Settings the array of flags is easier than splitting the fd list.